peachjam/models/judgment.py

```diff
@@ -86,12 +94,16 @@ def __str__(self):
     def assign_mnc(self):
         """Assign an MNC to this judgment, if one hasn't already been assigned."""
         if self.date and hasattr(self, "court"):
+            self.serial_number = self.generate_serial_number()
             if self.mnc != self.generate_citation():
-                self.serial_number = self.generate_serial_number()
```

Comment:
@longhotsummer This allows new mnc to be generated on change of serial number override.